test/suite/test_compact02.py

```diff
@@ -115,11 +115,9 @@ def test_compact02(self):
         c.close()
         self.pr('Removed total ' + str((count * 9666) / mb) + 'MB')
 
-        # 4. Reopen the connection to force the file to disk (if we leave dirty
-        # blocks in the cache, it can affect how compact works depending on how
-        # and when those blocks are written, because they're being written best
-        # fit, not first-fit, as compaction does).
-        self.reopen_conn()
+        # 4. Checkpoint to make the deleted records available for reuse, so
+        # compaction can be successful.
+        self.session.checkpoint()
```

Comment:
I don't think this is necessary.  The test runs without it (I ran 50 iterations on the AWS box that was failing reliably).  The code in `compact_file` calls checkpoint before starting and on each pass already.  I also think not having it is more realistic for what a user is going to do.  Did you see a failure without it?